sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationSettingJsonDeserializer.java

```diff
@@ -199,7 +190,19 @@ private static FeatureFlagConfigurationSetting readFeatureFlagConfigurationSetti
         return setting;
     }
 
-    private static FeatureFlagConfigurationSetting readFeatureFlagConfigurationSettingValue(String settingValue) {
+    public static SecretReferenceConfigurationSetting readSecretReferenceConfigurationSettingValue(String key,
```

Comment:
This method (and other methods below) is now public and should have javadoc (even if it's in the implementation package).